src/rust/ide/view/graph-editor/src/component/node/vcs.rs

```diff
@@ -0,0 +1,193 @@
+
+            (outer-inner).fill(color_rgba).into()
+        }
+    }
+}
+
+
+
+// ==============================
+// === Status Indicator Model ===
+// ==============================
+
+/// Internal data of `StatusIndicator`.
+#[derive(Clone,CloneRef,Debug)]
+struct StatusIndicatorModel {
+    shape : ShapeView<status_indicator_shape::Shape>,
+    root  : display::object::Instance,
+}
+
+impl StatusIndicatorModel {
+    fn new(logger: &Logger, app:&Application) -> Self {
+        let scene  = app.display.scene();
+        let shape = ShapeView::<status_indicator_shape::Shape>::new(&logger,scene);
+        let root = display::object::Instance::new(&logger);
+        root.add_child(&shape);
+        StatusIndicatorModel{shape, root}
+    }
+
+    fn hide(&self) {
+        self.shape.unset_parent();
+    }
+
+    fn show(&self) {
+        self.root.add_child(&self.shape);
+    }
+}
+
+impl display::Object for StatusIndicatorModel {
+    fn display_object(&self) -> &display::object::Instance {
+        &self.root
+    }
+}
+
+// ========================
+// === Status Indicator ===
+// ========================
+
+ensogl::define_endpoints! {
+    Input {
+        set_status     (Status),
+        set_size       (Vector2),
+        set_visibility (bool),
+    }
+    Output {
+        status (Status),
+    }
+}
+
+#[derive(Clone,CloneRef,Debug)]
+#[allow(missing_docs)]
+pub struct StatusIndicator {
+        model : Rc<StatusIndicatorModel>,
+    pub frp   : Frp,
+}
+
+impl StatusIndicator {
+    /// Constructor.
+    pub fn new(app:&Application) -> Self {
+        let logger = Logger::new("status_indicator");
+        let model  = Rc::new(StatusIndicatorModel::new(&logger,app));
+        let frp    = Frp::new();
+        Self {frp,model}.init_frp(app)
+    }
+
+    fn init_frp(self, app:&Application) -> Self {
+        let frp             = &self.frp;
+        let model           = &self.model;
+        let network         = &frp.network;
+        let indicator_color = color::Animation::new(network);
+
+        // FIXME : StyleWatch is unsuitable here, as it was designed as an internal tool for shape
+        // system (#795)
+        let styles = StyleWatch::new(&app.display.scene().style_sheet);
+
+        frp::extend! { network
+            frp.source.status <+ frp.input.set_status;
+
+            status_color <- frp.set_status.map(f!([styles](status) {
+                status.get_highlight_color_from_style(&styles)
+            }));
+            indicator_color.target <+ status_color;
+
+            eval indicator_color.value ((c)
+                model.shape.shape.color_rgba.set(color::Rgba::from(c).into())
+            );
+
+            eval frp.input.set_size ((size)
+                model.shape.shape.sprite.size.set(*size);
+            );
+
+            eval frp.input.set_visibility ([model](visible)
+                if *visible { model.show() } else { model.hide() }
+            );
```

Comment:
use `model.set_visibility(visibility)` instead - this will make FRP definition shorter and easier to read.